freqtrade/optimize/backtesting.py

```diff
@@ -691,15 +693,45 @@ def start(self) -> None:
         self.load_bt_data_detail()
         logger.info("Dataload complete. Calculating indicators")
 
+        run_ids = {
+            strategy.get_strategy_name(): get_strategy_run_id(strategy)
+            for strategy in self.strategylist
+        }
+
+        # Load previous result that will be updated incrementally.
+        if not self.config.get('no_backtest_cache', False):
+            self.results = find_existing_backtest_stats(
+                self.config['user_data_dir'] / 'backtest_results', run_ids)
+
         for strat in self.strategylist:
+            if self.results and strat.get_strategy_name() in self.results['strategy']:
+                # When previous result hash matches - reuse that result and skip backtesting.
+                logger.info(f'Reusing result of previous backtest for {strat.get_strategy_name()}')
+                continue
             min_date, max_date = self.backtest_one_strategy(strat, data, timerange)
-        if len(self.strategylist) > 0:
 
-            self.results = generate_backtest_stats(data, self.all_results,
-                                                   min_date=min_date, max_date=max_date)
+        # Update old results with new ones.
+        if len(self.all_results) > 0:
+            results = generate_backtest_stats(
+                data, self.all_results, min_date=min_date, max_date=max_date)
+            if self.results:
+                self.results['metadata'].update(results['metadata'])
+                self.results['strategy'].update(results['strategy'])
+                self.results['strategy_comparison'].extend(results['strategy_comparison'])
+            else:
+                self.results = results
 
             if self.config.get('export', 'none') == 'trades':
                 store_backtest_stats(self.config['exportfilename'], self.results)
```

Comment:
Just for my understanding, maybe ...
assuming i backtested `StrategyA` yesterday, and today i use `--strategy-list StrategyA StrategyB` - (nothing else changed, so `StrategyA` is loaded from the prior result) ... 

do we intentionally write the result of StrategyA to disk again? (so it's now in 2 result files)

Reply:
I suspect `backtesting-show` would break if we did not copy old backtest result to new result file. But more importantly copying keeps results consistent in the event where user may delete some but not all old backtest results. If it were the case, in example you described user may lose half of the result even though file of that backtest result is present.